zebra-chain/src/equihash_solution.rs

```diff
@@ -20,11 +20,13 @@ const EQUIHASH_SOLUTION_SIZE: usize = 1344;
 ///
 /// The size of an Equihash solution in bytes is always 1344 so the
 /// length of this type is fixed.
-pub struct EquihashSolution(pub [u8; EQUIHASH_SOLUTION_SIZE]);
+pub struct EquihashSolution(pub Box<[u8; EQUIHASH_SOLUTION_SIZE]>);
 
 impl PartialEq<EquihashSolution> for EquihashSolution {
     fn eq(&self, other: &EquihashSolution) -> bool {
-        self.0.as_ref() == other.0.as_ref()
+        let unboxed_self = *(self.0);
+        let unboxed_other = *(other.0);
+        unboxed_self.as_ref() == unboxed_other.as_ref()
```

Comment:
@hdevalence is there a prettier way to do this than literally unboxing them to make the compiler happy?

Reply:
Hmm, `self.0.as_ref() == other.0.as_ref()` doesn't work? Is that because `Box<[u8; 1344]>: AsRef<[u8; 1344]>` but `[u8; 1344]` doesn't implement `Eq`?  If so, would it work to do
```
self.0.as_ref()[..] == other.0.as_ref()[..]
```
to convert the `[u8; 1344]` to `[u8]`?